samples/boards/96b_argonkey/microphone/README.rst

```diff
@@ -0,0 +1,131 @@
+.. _ArgonKey_Microphone:
+
+ArgonKey Board Microphone
+#########################
+
+Overview
+********
+This sample provides an example of how to acquire audio through
+the on-board MP34DT05 microphone. The microphone generates a PDM
+stream which is acquired through I2S. The PDM stream is then
+converted to PCM using the OpenPDMFilter library, which is available
+in source code in this sample.
+
+Requirements
+************
+
+This sample just requires the ArgonKey board. The board can be fed
+in either one of the following two ways:
+
+- mezzanine mode, plugging the ArgonKey to HiKey board thru its 96Board
+  low-speed connector
+- standalone mode, feeding directly 5V on P1 connector
+
+References
+**********
+
+- :ref:`96b_argonkey`
+
+Building and Running
+********************
+
+ .. zephyr-app-commands::
+    :zephyr-app: samples/boards/96b_argonkey/microphone
+    :host-os: unix
+    :board: 96b_argonkey
+    :goals: run
+    :compact:
+
+Sample Output
+=============
+
+The example acquires 3s of audio and prints out the PCM stream on COM port.
+A USB to TTL 1V8 serial cable may be attached to the low speed connector on
+the back of the board on P3.5 (TX) and P3.7 (RX).
+
+As soon as the acquisition starts the green led glows. At the end of the
+acquisition both the green and red led goes on and the PCM stream is sent
+on COM port. When the output is completed the green led goes off and the red
+led stays on. The process can be re-iterated only by pressing the RST button.
+
+- acquisition starts:     GRN on  - RED off
+- acquisition ends:       GRN on  - RED on
+- output ends:            GRN off - RED on
+
+The characteristic of the PCM audio are hardcoded in the example:
+
+- 32KHz of sample rate
+- 16 bits per sample
+- 1 channel (mono), as only 1 microphone is available
+
+3 seconds of acquisition @32Kz of sampling rate corresponds to 96000 sample
+(x 16bits each).
+
+Since the MP34DT05 microphone is PDM, we need to apply an oversampling factor,
+which is 64 in this example, resulting in a 2MHz bit clock. During the PCM
+conversion the OpenPDMFilter library is filtering out and decimating the data
+by the same 64 factor.
+
+At the end of the acquisition the PCM data will be printed on the terminal
+emulator in either binary or ascii format. The output is controlled by
+PCM_OUTPUT_IN_ASCII definition, which is off by default.
+
+Binary PCM Output
+-----------------
+
+The ttyUSB0 port must be configured in raw mode in order to avoid to have
+characters 'cooked' out.
+
+ .. code-block:: console
+
+    stty -F /dev/ttyUSB0 115200 raw
+    cat /dev/ttyUSB0 > /tmp/sound.raw
+
+NOTE!
```

Comment:
Change this to:

    ..note:: In case the character 0x0a is interpreted as NL and an 0x0d (CR) is added,
       you may need to remove it::

          dos2unix -f /tmp/sound.raw